python/mujinplanningclient/binpickingplanningclient.py

```diff
@@ -46,641 +75,2953 @@ def __init__(self, regionname=None, **kwargs):
+        if deleteDynamicObjectsAfterFinish != False:
+            taskparameters['deleteDynamicObjectsAfterFinish'] = deleteDynamicObjectsAfterFinish
+        if deleteTargetWhenPlacedInDest != 'KeepInAll':
+            taskparameters['deleteTargetWhenPlacedInDest'] = deleteTargetWhenPlacedInDest
+        if deleteTargetFromSourceContainer != 0:
+            taskparameters['deleteTargetFromSourceContainer'] = deleteTargetFromSourceContainer
+        if deletetargetonocclusion != True:
+            taskparameters['deletetargetonocclusion'] = deletetargetonocclusion
+        if destikparamnames is not None:
+            taskparameters['destikparamnames'] = destikparamnames
+        if intermediateCycles is not None:
+            taskparameters['intermediateCycles'] = intermediateCycles
+        if deleteTargetsOnPieceLost is not None:
+            taskparameters['deleteTargetsOnPieceLost'] = deleteTargetsOnPieceLost
+        if deleteTargetsOnRecovery is not None:
+            taskparameters['deleteTargetsOnRecovery'] = deleteTargetsOnRecovery
+        if deleteTargetsEveryCycleSlowMode is not None:
+            taskparameters['deleteTargetsEveryCycleSlowMode'] = deleteTargetsEveryCycleSlowMode
+        if destBarcodeScanningInfo is not None:
+            taskparameters['destBarcodeScanningInfo'] = destBarcodeScanningInfo
+        if destBarcodeScanningInfoPerContainer is not None:
+            taskparameters['destBarcodeScanningInfoPerContainer'] = destBarcodeScanningInfoPerContainer
+        if destTargetAABBAlignIkParameters is not None:
+            taskparameters['destTargetAABBAlignIkParameters'] = destTargetAABBAlignIkParameters
+        if destTargetAnyBottomFaceRotationParameters is not None:
+            taskparameters['destTargetAnyBottomFaceRotationParameters'] = destTargetAnyBottomFaceRotationParameters
+        if destTargetCornerParameters is not None:
+            taskparameters['destTargetCornerParameters'] = destTargetCornerParameters
+        if destTargetStackParameters is not None:
+            taskparameters['destTargetStackParameters'] = destTargetStackParameters
+        if dynamicGoalsParameters is not None:
+            taskparameters['dynamicGoalsParameters'] = dynamicGoalsParameters
+        if ensurePickVisibilityAtPlace != 'preferred':
+            taskparameters['ensurePickVisibilityAtPlace'] = ensurePickVisibilityAtPlace
+        if graspApproachInfos is not None:
+            taskparameters['graspApproachInfos'] = graspApproachInfos
+        if graspApproachInfosPerURI is not None:
+            taskparameters['graspApproachInfosPerURI'] = graspApproachInfosPerURI
+        if graspDepartInfos is not None:
+            taskparameters['graspDepartInfos'] = graspDepartInfos
+        if graspDepartAboveNearbyObstacles != False:
+            taskparameters['graspDepartAboveNearbyObstacles'] = graspDepartAboveNearbyObstacles
+        if graspDepartAboveNearbyObstaclesMaxDist != 600:
```

Comment:
Same for all of these - the default set by the server can be different from the default that the client used to set. For this parameter, the server sets no default so it would be empty.

Reply:
Updated.

Reply:
Ah, sorry, this one isn't ready.

Reply:
Discussed offline: the default values would need more investigation to understand what the server actually wants.

On the other hand, there are parameters that are currently always passed to the server. We will recover that functionality.

Reply:
This one looks fine.